File: 백준/Silver/10866. 덱/덱.py
import sys
from collections import deque

N = int(sys.stdin.readline().rstrip())

deq = deque()
def deque_check(order):
    if 'push' in order:
        order_i, k = order.split(' ')
        if order_i == 'push_front':
            deq.appendleft(int(k))
        elif order_i == 'push_back':
            deq.append(int(k))
    elif order == 'pop_front':
        if not deq:
            return print(-1)
        k = deq.popleft()
        return print(k)
    elif order == 'pop_back':
        if not deq:
            return print(-1)
        k = deq.pop()
        return print(k)
    elif order == 'size':
        return print(len(deq))
    elif order == 'empty':
        return print(0 if deq else 1)
    elif order == 'front':
        return print(deq[0] if deq else -1)
    elif order == 'back':
        return print(deq[-1] if deq else -1)


for _ in range(N):
    order = sys.stdin.readline().rstrip()
    deque_check(order)

File: 백준/Silver/10866. 덱/덱.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		
		int N = Integer.parseInt(br.readLine());
		
		Deque<Integer> queue = new ArrayDeque<>();
		
		for (int i=0;i<N;i++) {
			StringTokenizer st = new StringTokenizer(br.readLine());
			
			String odr = st.nextToken();
			
			if (odr.equals("push_back")) {
				queue.addLast(Integer.parseInt(st.nextToken()));
			} else if (odr.equals("push_front")) {
				queue.addFirst(Integer.parseInt(st.nextToken()));
			} else if (odr.equals("pop_front")) {
				sb.append(queue.isEmpty()? -1:queue.removeFirst()).append("\n");
			} else if (odr.equals("pop_back")) {
				sb.append(queue.isEmpty()? -1:queue.removeLast()).append("\n");
			} else if (odr.equals("size")) {
				sb.append(queue.size()).append("\n");
			} else if (odr.equals("empty")) {
				sb.append(queue.isEmpty()? 1:0).append("\n");
			} else if (odr.equals("front")) {
				sb.append(queue.isEmpty()? -1:queue.peekFirst()).append("\n");
			} else if (odr.equals("back")) {
				sb.append(queue.isEmpty()? -1:queue.peekLast()).append("\n");
			}

		}
		
		System.out.println(sb);


		
	}
}
